Flutter Run Error

### Search before asking

- [x] I have searched https://github.com/ultralytics/yolo-flutter-app/issues and did not find a similar report.


### Project area

Android build

### Bug

Getting an error while running in Flutter,

Build.gradle:
ndkVersion = "28.2.13676358"

pubspec.yaml
  # AI Inference
  ultralytics_yolo:
    git:
      url: https://github.com/ultralytics/ultralytics_yolo.git
      ref: main
  logger: ^2.6.2


FAILURE: Build failed with an exception.                                      
                                                                                
  * What went wrong:                                                            
  Execution failed for task                                                     
  ':ultralytics_yolo:buildCMakeRelWithDebInfo[arm64-v8a]'.                      
  > com.android.ide.common.process.ProcessException: ninja: Entering directory  
  `/home/e-alpha/.pub-cache/git/ultralytics_yolo-1fd9a1c4f582078ddb601b3e82f62c 
  ad93bf3313/android/.cxx/RelWithDebInfo/27492q1m/arm64-v8a'                    
    [1/2] Building CXX object CMakeFiles/ultralytics.dir/native-lib.cpp.o       
    [2/2] Linking CXX shared library "/home/e-alpha/Documents/Git/Device Smart  
  Vision/smart_vision_device_app/build/ultralytics_yolo/intermediates/cxx/RelWi 
  thDebInfo/27492q1m/obj/arm64-v8a/libultralytics.so"                           
    FAILED: /home/e-alpha/Documents/Git/Device Smart                            
  Vision/smart_vision_device_app/build/ultralytics_yolo/intermediates/cxx/RelWi 
  thDebInfo/27492q1m/obj/arm64-v8a/libultralytics.so                            
    : && /home/e-alpha/Android/Sdk/ndk/28.2.13676358/toolchains/llvm/prebuilt/l 
  inux-x86_64/bin/clang++ --target=aarch64-none-linux-android21                 
  --sysroot=/home/e-alpha/Android/Sdk/ndk/28.2.13676358/toolchains/llvm/prebuil 
  t/linux-x86_64/sysroot -fPIC -g -DANDROID -fdata-sections -ffunction-sections 
   -funwind-tables -fstack-protector-strong -no-canonical-prefixes              
  -D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -frtti -fexceptions      
  -std=c++17 -fexceptions -frtti -O2 -g -DNDEBUG  -Wl,--build-id=sha1           
  -Wl,--no-rosegment -Wl,--no-undefined-version -Wl,--fatal-warnings            
  -Wl,--no-undefined -Qunused-arguments  -Wl,--gc-sections -shared              
  -Wl,-soname,libultralytics.so -o "/home/e-alpha/Documents/Git/Device Smart    
  Vision/smart_vision_device_app/build/ultralytics_yolo/intermediates/cxx/RelWi 
  thDebInfo/27492q1m/obj/arm64-v8a/libultralytics.so"                           
  CMakeFiles/ultralytics.dir/native-lib.cpp.o                                   
  /home/e-alpha/Android/Sdk/ndk/28.2.13676358/toolchains/llvm/prebuilt/linux-x8 
  6_64/sysroot/usr/lib/aarch64-linux-android/21/liblog.so  -latomic -lm && :    
    ld.lld: error: undefined symbol: std::__throw_length_error(char const*)     
    >>> referenced by stl_vector.h:1756                                         
  (/snap/flutter/current/usr/include/c++/9/bits/stl_vector.h:1756)              
    >>>               CMakeFiles/ultralytics.dir/native-lib.cpp.o:(Java_com_ult 
  ralytics_yolo_ObjectDetector_postprocess)                                     
    >>> referenced by stl_vector.h:1756                                         
  (/snap/flutter/current/usr/include/c++/9/bits/stl_vector.h:1756)              
    >>>               CMakeFiles/ultralytics.dir/native-lib.cpp.o:(Java_com_ult 
  ralytics_yolo_ObjectDetector_postprocess)                                     
    >>> referenced by stl_vector.h:1767                                         
  (/snap/flutter/current/usr/include/c++/9/bits/stl_vector.h:1767)              
    >>>               CMakeFiles/ultralytics.dir/native-lib.cpp.o:(Java_com_ult 
  ralytics_yolo_ObjectDetector_postprocess)                                     
    >>> referenced 3 more times                                                 
    clang++: error: linker command failed with exit code 1 (use -v to see       
  invocation)                                                                   
    ninja: build stopped: subcommand failed.                                    
                                                                                
    C++ build system [build] failed while executing:                            
        /home/e-alpha/Android/Sdk/cmake/3.22.1/bin/ninja \                      
          -C \                                                                  
          /home/e-alpha/.pub-cache/git/ultralytics_yolo-1fd9a1c4f582078ddb601b3 
  e82f62cad93bf3313/android/.cxx/RelWithDebInfo/27492q1m/arm64-v8a \            
          ultralytics                                                           
      from /home/e-alpha/.pub-cache/git/ultralytics_yolo-1fd9a1c4f582078ddb601b 
  3e82f62cad93bf3313/android                                                    
                                                                                
  * Try:                                                                        
  > Run with --stacktrace option to get the stack trace.                        
  > Run with --info or --debug option to get more log output.                   
  > Run with --scan to get full insights.                                       
  > Get more help at https://help.gradle.org.                                   
                                                                                
  BUILD FAILED in 1m 10s                                                        
  Running Gradle task 'assembleRelease'...                           70.7s      
  Error: Gradle task assembleRelease failed with exit code 1                    
  #0      throwToolExit (package:flutter_tools/src/base/common.dart:34:3)       
  #1      AndroidGradleBuilder.buildGradleApp                                   
  (package:flutter_tools/src/android/gradle.dart:592:7)                         
  <asynchronous suspension>                                                     
  #2      AndroidGradleBuilder.buildApk                                         
  (package:flutter_tools/src/android/gradle.dart:234:5)                         
  <asynchronous suspension>                                                     
  #3      AndroidDevice.startApp                                                
  (package:flutter_tools/src/android/android_device.dart:568:7)                 
  <asynchronous suspension>                                                     
  #4      FlutterDevice.runCold                                                 
  (package:flutter_tools/src/resident_runner.dart:572:33)                       
  <asynchronous suspension>                                                     
  #5      ColdRunner.run (package:flutter_tools/src/run_cold.dart:56:28)        
  <asynchronous suspension>                                                     
  #6      RunCommand.runCommand                                                 
  (package:flutter_tools/src/commands/run.dart:904:26)                          
  <asynchronous suspension>                                                     
  #7      FlutterCommand.run.<anonymous closure>                                
  (package:flutter_tools/src/runner/flutter_command.dart:1590:27)               
  <asynchronous suspension>                                                     
  #8      AppContext.run.<anonymous closure>                                    
  (package:flutter_tools/src/base/context.dart:154:19)                          
  <asynchronous suspension>                                                     
  #9      CommandRunner.runCommand (package:args/command_runner.dart:212:13)    
  <asynchronous suspension>                                                     
  #10     FlutterCommandRunner.runCommand.<anonymous closure>                   
  (package:flutter_tools/src/runner/flutter_command_runner.dart:496:9)          
  <asynchronous suspension>                                                     
  #11     AppContext.run.<anonymous closure>                                    
  (package:flutter_tools/src/base/context.dart:154:19)                          
  <asynchronous suspension>                                                     
  #12     FlutterCommandRunner.runCommand                                       
  (package:flutter_tools/src/runner/flutter_command_runner.dart:431:5)          
  <asynchronous suspension>                                                     
  #13     FlutterCommandRunner.run.<anonymous closure>                          
  (package:flutter_tools/src/runner/flutter_command_runner.dart:307:33)         
  <asynchronous suspension>                                                     
  #14     run.<anonymous closure>.<anonymous closure>                           
  (package:flutter_tools/runner.dart:104:11)                                    
  <asynchronous suspension>                                                     
  #15     AppContext.run.<anonymous closure>                                    
  (package:flutter_tools/src/base/context.dart:154:19)                          
  <asynchronous suspension>                                                     
  #16     main (package:flutter_tools/executable.dart:103:3)                    
  <asynchronous suspension>          

### Environment

Dart Version: 3.11.5
Flutter Verion: Flutter 3.41.7
Android Studio Meerkat | 2024.3.1
OS: Ubuntu 24.04.2 LTS

### Minimal Reproducible Example

_No response_

### Additional

_No response_

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

--- a/android/src/main/cpp/CMakeLists.txt
+++ b/android/src/main/cpp/CMakeLists.txt
@@ -39,6 +39,6 @@ log
 
 target_link_libraries(
 ultralytics
+c++_shared
 ${log-lib}
 )
-

